Profile

Hydravlick opened the pull request.

Hydravlick committed:
First attempt

Commit:
fix: mysql connection from php

Hydravlick committed:
Second day

Hydravlick committed:
Third day

Hydravlick committed:
Сегодня изменил роут с case на elseif. Добавил файл clients.php в котором привёл массив к типу JSON и вывел в данные из него на странице

Hydravlick committed:
промежуточный коммит 27.
поправить

Hydravlick committed:
240427
Cоздал свой первый класс в yii2, что выводит таблицу пользователей на экран с помощью GridView.
Полностью удалил бд и создал шаблон миграцией.
С помощю бд смог аудентифицироваться как админ в backend.test

Hydravlick committed:
2022.04.02

Чистая версия AdminLTE

Hydravlick committed:
2022.04.11

Hydravlick committed:
2022.04.11

Created profile.php and change index.php

Hydravlick committed:
2022.04.11

Created profile.php and change index.php

Hydravlick committed:
2022.04.11

Changed version

Hydravlick committed:
2022.04.11

Changed version 1.1

GlebDresv commented on app/common/models/UserAuth.php:
```diff
@@ -0,0 +1,84 @@
+ *
+ * @property User $user
+ */
+class UserAuth extends \yii\db\ActiveRecord
```
UserProfile?
явно модель к авторизации не относиться
название не удачное

GlebDresv commented on app/frontend/views/site/index.php:
```diff
@@ -0,0 +1,31 @@
+use yii\grid\GridView;
+
+/** @var yii\web\View $this
+ * @var $dataProvider UserAuth[]
```
$dataProvider  не масив объектов UserAuth
перепроверь и напиши правду)

GlebDresv commented on app/frontend/controllers/SiteController.php:
```diff
@@ -0,0 +1,282 @@
+     *
+     * @return mixed
+     */
+    public function actionProfile()
```
Это стоит вынести в отдельный контроллер - profileController
где будут происходить действия с профилем

GlebDresv commented on app/frontend/controllers/SiteController.php:
```diff
@@ -0,0 +1,282 @@
+     */
+    public function actionProfile()
+    {
+        $personal_info = User::find()->where('username' == Yii::$app->user->identity->username)->one();
```
чекни что лежит в переменной Yii::$app->user->identity

GlebDresv replied:
camelCase

GlebDresv replied:
изза магии внутри фреймворка шторм не понимает что у тебя лежит в переменной $personal_info
ему надо подсказать аннотацией
/** @var ClassName $personal_info */

GlebDresv commented on app/common/models/User.php:
```diff
@@ -0,0 +1,218 @@
+    {
+        $this->password_reset_token = null;
+    }
+    public function getPublicinfo()
```
+ camelCase
+ метод возвращает нечто UserAuth а называется getPublicinfo - нестыковочка
+ надо добавить корректный @property  в аннотацию класса на этот гетер

GlebDresv commented on app/common/models/UserAuth.php:
```diff
@@ -0,0 +1,84 @@
+ * @property int $desired_distance
+ * @property int $recommended_distance
+ *
+ * @property User $user
```
Попробуй получить это свойство класса, помоему тут такого нет

GlebDresv commented on app/common/models/UserAuth.php:
```diff
@@ -0,0 +1,84 @@
+     *
+     * @return \yii\db\ActiveQuery
+     */
+    public function getPrivateinfo()
```
метод возвращает нечто User а называется getPrivateinfo - нестыковочка

GlebDresv commented on app/frontend/controllers/SiteController.php:
```diff
@@ -0,0 +1,282 @@
+    public function actionProfile()
+    {
+        $personal_info = User::find()->where('username' == Yii::$app->user->identity->username)->one();
+        $public_info = UserAuth::find()->where('userid' == Yii::$app->user->id)->one();
```
$public_info = $personal_info->publicinfo;

GlebDresv replied:
camelCase

GlebDresv commented on app/frontend/views/site/profile.php:
```diff
@@ -0,0 +1,27 @@
+
+/**
+ * @var frontend\controllers\SiteController $personal_info
+ * @var frontend\controllers\SiteController $public_info
```
вранье
ни $personal_info
ни $public_info

не являються инстансами класса  frontend\controllers\SiteController
кстати тебя не смутило когда у тебя автодополнение перестало работать в шторме?

Hydravlick committed:
2022.04.11

Исправил миграции, добавил папку db в .gitignore

Hydravlick committed:
2022.04.11

Hydravlick committed:
2022.04.11

fixed migrations

GlebDresv commented on app/console/migrations/m130524_201442_public.php:
```diff
@@ -0,0 +1,40 @@
+/**
+ * Class m220427_174111_people
+ */
+class m130524_201442_public extends Migration
```
лучше именовать миграции так что б было понятно что в них происходит без необходимости смотреть в их код

например
m130524_201442_create_table_public_info

GlebDresv replied:
`CREATE TABLE table_name`
https://www.w3schools.com/sql/sql_create_table.asp

`ALTER TABLE table_name
ADD column_name`
https://www.w3schools.com/sql/sql_alter.asp


именно поэтому лучше для названия миграции в которой добавляется таблица использовать слово create а не add

GlebDresv commented on app/console/migrations/m130524_201442_public.php:
```diff
@@ -0,0 +1,40 @@
+        $this->createTable('public_info', [
+            'userid' => $this->primaryKey(),
+            'phone_number' => $this->integer()->defaultValue(Null),
+            'full_name' => $this->string()->notNull()->defaultValue('No Name'),
```
представь регаешься ты на сайте а тебе в имя записывают ноНейм
не самое ожидаемое поведение для многих, в данном случае лучше дефоултом пустую строку вставлять

GlebDresv commented on app/console/migrations/m130524_201442_public.php:
```diff
@@ -0,0 +1,40 @@
+            'userid' => $this->primaryKey(),
+            'phone_number' => $this->integer()->defaultValue(Null),
+            'full_name' => $this->string()->notNull()->defaultValue('No Name'),
+            'about' => $this->string()->notNull()->defaultValue(Null),
```
стремота какаята написана) опять таки, лучше пустая строка

GlebDresv commented on app/frontend/controllers/ProfileController.php:
```diff
@@ -0,0 +1,29 @@
+         * @var $personal_info User;
+         *
+         */
+        $personal_info = Yii::$app->user->identity;
```
camelCase

GlebDresv commented on app/frontend/controllers/ProfileController.php:
```diff
@@ -0,0 +1,29 @@
+     *
+     * @return mixed
+     */
+    public function actionIndex()
```
/profile  вернет 404
/profile/index вернет ошибку для незарегистрированного пользователя
/profile/index вернет профиль пользователя для пользователя который зарегистрирован

вопрос на засыпку, как посмотреть профиль другого пользователя?
 не свой)

GlebDresv commented on app/frontend/controllers/SiteController.php:
```diff
@@ -0,0 +1,268 @@
+     */
+    public function actionIndex()
+    {
+        $user = PublicInfo::find()->all();
```
что будем делать когда на сайте зарегистрируется 99999 участников?))
представь себе какой будет результат в таком случае

GlebDresv commented on app/frontend/controllers/ProfileController.php:
```diff
@@ -0,0 +1,29 @@
+use common\models\User;
+use Yii;
+
+class ProfileController extends \yii\web\Controller
```
use yii\web\Controller;

 ....   extends Controller

GlebDresv commented on app/frontend/views/profile/index.php:
```diff
@@ -0,0 +1,26 @@
+?>
+<div class="site-profile">
+    <h1><?= Html::encode($this->title) ?></h1>
+    <img src="<?= $public_info->image ?>" width="220px" height="220px" border-radius="50%" alt="">
```
мне тут ide подсвечивает красным $public_info

GlebDresv commented on app/frontend/views/site/index.php:
```diff
@@ -0,0 +1,32 @@
+        <?= GridView::widget([
+            'dataProvider' => $dataProvider,
+            'columns' => [
+                'full_name',
```
давай добавим переход в профиль пользователя по клику на его имя в таблице

GlebDresv commented on app/frontend/views/site/index.php:
```diff
@@ -0,0 +1,32 @@
+
+/**
+ * @var yii\web\View $this
+ * @var $dataProvider PublicInfo
```
опять вранье $dataProvider не является инстансом класса PublicInfo
если не знаешь - узнай, угадать не получиться

Hydravlick committed:
2022.04.11

fixed migrations again

Hydravlick committed:
2022.04.11

fixed migrations again 1.1

Hydravlick committed:
2022.04.11

add show.php and updated ProfileController.php
now it`s possible to dynamically call profile by id

Hydravlick committed:
2022.04.12

21:15

Hydravlick committed:
2022.04.12

better method

Hydravlick committed:
2022.04.12

better method

Hydravlick committed:
2022.04.12

21 24

GlebDresv commented on app/frontend/controllers/ProfileController.php:
```diff
@@ -0,0 +1,52 @@
+        /**
+         * @var $userInfo User;
+         */
+        $userInfo = User::findOne(['id' => $id]);
+        if ($userInfo != null) {
+            $profileInfo = $userInfo->publicInfo;
+            return $this->render('show', [
+                'userInfo' => $userInfo,
+                'profileInfo' => $profileInfo,
+            ]);
+        } else {
+            throw new NotFoundHttpException();
+        }
```
```suggestion
        if(!$userInfo){
            throw new NotFoundHttpException();
        }
        // все остальное
```

GlebDresv commented on app/frontend/views/site/index.php:
```diff
@@ -0,0 +1,35 @@
+
+/**
+ * @var yii\web\View $this
+ * @var ArrayDataProvider $dataProvider
```
да что ж такое... опять не угадал

GlebDresv commented on app/frontend/views/profile/index.php:
```diff
@@ -0,0 +1,32 @@
+/**
+ *
+ * @var yii\web\View $this
+ * @var User $userInfo
```
странно класс User, a переменная называется $userInfo...
искать ли сакральный смысл в этом переименовании?

GlebDresv commented on app/frontend/views/profile/index.php:
```diff
@@ -0,0 +1,32 @@
+<div class="profile-index">
+    <h1><?= Html::encode($this->title) ?></h1>
+    <img src="<?= $profileInfo->image ?>" width="220px" height="200px" border-radius="50%" alt="">
+    <p><b><?= $userInfo->username . ' (' . $profileInfo->full_name . ')' ?></b>
```
давай как-то избавимся от пустых скобочек
https://ibb.co/FJ9k83Z

GlebDresv commented on app/frontend/controllers/SiteController.php:
```diff
@@ -0,0 +1,267 @@
+     */
+    public function actionIndex()
+    {
+        $user = ProfileInfo::find()->orderBy('distance_all')->limit(100);
```
близко, но неужели мы хотим показать топ 100 неудачников?)))

Hydravlick committed:
2022.04.12

сделал ссылку в таблице и исправил ошибки

Hydravlick committed:
2022.04.12

last iteration
